langchain/tools/plugin.py

```diff
@@ -1,28 +1,56 @@
 from __future__ import annotations
 
+from typing import Optional
+
 import requests
+from pydantic import BaseModel
 
 from langchain.tools.base import BaseTool
 
 
+class ApiConfig(BaseModel):
+    type: str
+    url: str
+    has_user_authentication: Optional[bool] = False
+
+
+class AIPlugin(BaseModel):
+    schema_version: str
+    name_for_model: str
+    name_for_human: str
+    description_for_model: str
+    description_for_human: str
+    auth: Optional[dict] = None
+    api: ApiConfig
+    logo_url: Optional[str]
+    contact_email: Optional[str]
+    legal_info_url: Optional[str]
+
+
 class AIPluginTool(BaseTool):
+    plugin: AIPlugin
     api_spec: str
 
     @classmethod
     def from_plugin_url(cls, url: str) -> AIPluginTool:
-        response = requests.get(url).json()
+        cls.plugin = AIPlugin(**requests.get(url).json())
```

Comment:
i would not set cls.plugin here (i think thats just confusing, and we pass it in down below, so probs not necc) but i can fix